File: MPP/proiect/java/persistence/src/main/java/repository/ShowDataSqlRepo.java
package repository;

import model.ShowData;
import utils.DbConnManager;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;
import java.util.Properties;

/**
 * Created by vitiv on 3/19/17.
 */
public class ShowDataSqlRepo implements ICrudRepository<ShowData, Integer> {
    private DbConnManager connManager;


    public ShowDataSqlRepo(Properties props) {
        connManager = new DbConnManager(props);
    }

    @Override
    public long size() {
        throw new UnsupportedOperationException();
    }

    @Override
    public void add(ShowData entity) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void delete(Integer integer) {
        throw new UnsupportedOperationException();
    }

    @Override
    public ShowData find(Integer integer) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Collection<ShowData> getAll() {
        Connection con = connManager.getConnection();
        ArrayList<ShowData> ShowDatas = new ArrayList<>();

        String queryStr = "select s.id as id, a.name as artistName,  " +
                "l.name as locationName, datetime(s.startTime) as startTime,  " +
                "cast(total(p.quantity) as integer) as soldSeats, s.availableSeats as availableSeats " +
                "from  Show as s  join Artist as a on a.id = s.artistId join Location as l " +
                "on l.id = s.locationId left join Purchase as p on p.showId = s.id group by s.id;";
        try (PreparedStatement preStmt = con.prepareStatement(queryStr)) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("id");
                    String artistName = result.getString("artistName");
                    String locationName = result.getString("locationName");

                    SimpleDateFormat parseD = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
                    String startTimeStr = result.getString("startTime");
                    Date startTime = parseD.parse(startTimeStr);

                    int soldSeats = result.getInt("soldSeats");
                    int availableSeats = result.getInt("availableSeats");

                    ShowDatas.add(new ShowData(id, artistName, locationName, startTime, soldSeats, availableSeats));
                }
            } catch (ParseException e) {
                throw new RepositoryException("date parse error");
            }
        } catch (SQLException ex) {
            throw new RepositoryException("Error DB " + ex);
        }
        return ShowDatas;
    }
}

File: MPP/proiect/c#/Festival/Persistence/Repository/ShowDataSqlRepo.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Festival.Model;

namespace Festival.Repository
{
    public class ShowDataSqlRepo: ICrudRepository<int, ShowData>
    {
        private string filename;

        public ShowDataSqlRepo(string filename)
        {
            this.filename = filename;
        }

        public ShowData findOne(int id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<ShowData> findAll()
        {
            IDbConnection con = DbUtils.getConnection(filename);
            IList<ShowData> shows = new List<ShowData>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select s.id as id, a.name as artistName,  " +
                    "l.name as locationName, datetime(s.startTime) as startTime,  " +
                    "cast(total(p.quantity) as integer) as soldSeats, s.availableSeats as availableSeats " +
                    "from  Show as s  join Artist as a on a.id = s.artistId join Location as l " +
                    "on l.id = s.locationId left join Purchase as p on p.showId = s.id group by s.id;";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        string artistName = dataR.GetString(1);
                        string locationName = dataR.GetString(2);
                        DateTime startTime = dataR.GetDateTime(3);
                        int remainingSeats = dataR.GetInt32(5) - dataR.GetInt32(4);

                        shows.Add(new ShowData
                        {
                            Id = id,
                            ArtistName = artistName,
                            LocationName = locationName,
                            StartTime = startTime,
                            RemainingSeats = remainingSeats
                        });
                    }
                }
            }

            return shows;
        }

        public void save(ShowData entity)
        {
            throw new NotImplementedException();
        }

        public void delete(int id)
        {
            throw new NotImplementedException();
        }

        public void update(ShowData old, ShowData entity)
        {
            throw new NotImplementedException();
        }
    }
}
